c/tskit/tables.h

```diff
@@ -2488,6 +2492,67 @@ nodes (and individuals and populations) reordered.
 int tsk_table_collection_subset(
     tsk_table_collection_t *self, tsk_id_t *nodes, tsk_size_t num_nodes);
 
+/**
+@brief Forms the node-wise union of two table collections.
+
+@rst
+Expands this table collection by adding the exclusive portions of another table
+collection to itself. The ``node_mapping`` encodes which nodes in ``other`` are
+equivalent to a node in ``self``. The positions in the ``node_mapping`` array
+correspond to node ids in ``other``, and the elements encode the equivalent
+node id in ``self`` or TSK_NULL if the node is exclusive to ``other``. Nodes
+that are exclusive ``other`` are added to ``self``, along with:
+
+1. Individuals which are new to ``self``.
+2. Edges whose parent or child are new to ``self``.
+3. Sites which were not present in ``self``.
+4. Mutations whose nodes are new to ``self``.
+
+Nodes that are added onto ``self`` are consistently assigned new
+populations. Therefore, the number of populations in ``self`` after the union
+operation will be equal to the sum of the number of populations in ``self``
+and ``other``.
+
+Note that this operation implicitly sorts, deduplicates sites and computes
+mutation parents in ``self``; operations which may change ``self`` even if
+``other`` is empty or if all nodes of ``other`` are shared with ``self``.
+
+**Options**:
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_NO_CHECK_SHARED
+    By default, union checks that the portion of shared history between
+    ``self`` and ``other``, as implied by ``node_mapping``, are indeed
+    equivalent. It does so by subsetting both ``self`` and ``other`` on the
+    equivalent nodes specified in ``node_mapping``, and then checking for
+    equality of the subsets. Note two TableCollections could contain equivalent
+    histories and fail this check, in which cases this check should be disabled.
```

Comment:
Under what circumstances? But this note will be obsolete once we can do a strict sort, right? We should clarify or delete this sentence.